recipe/meta.yaml

```diff
@@ -215,18 +217,20 @@ outputs:
         - ninja
         # Keep libprotobuf here so that a compatibile version
         # of protobuf is installed between build and host
-        - libprotobuf
-        - protobuf
+        # TODO(baszalmstra): There are linker errors on Windows when using protobuf from conda-forge
+        - libprotobuf  # [not win]
+        - protobuf     # [not win]
         - make      # [linux]
       host:
         # GPU requirements
         - cudnn                           # [cuda_compiler_version != "None"]
-        - nccl                            # [cuda_compiler_version != "None"]
+        - nccl                            # [cuda_compiler_version != "None" and linux]
         - magma                           # [cuda_compiler_version != "None"]
         - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
         {% if cuda_major >= 12 %}
-        - cuda-driver-dev
+        - cuda-driver-dev                 # [linux]
         - cuda-cudart-dev
+        - cuda-crt
```

Comment:
I added this to try and fix this build issue (from this run):

```
C:\bld\libtorch_1720613693972\_h_env\Library\include\cuda_runtime.h(83): fatal error C1083: Cannot open include file: 'crt/host_config.h': No such file or directory
```

However, adding that package fails because it seems it cant resolve.

This file is added by `cuda-nvcc-dev_win-64` but its only added in the build environment. `cuda-crt` however, seems to only be available from cuda 12.2. Would it be appropriate to depend on `cuda-nvcc-dev_win-64` in the `host` section for lower cuda versions?

Reply:
And why should it be restricted to `linux` only? It seems to also be available on windows? https://prefix.dev/channels/conda-forge/packages/cuda-crt

Reply:
This was from me looking at the [CI](https://github.com/conda-forge/pytorch-cpu-feedstock/actions/runs/9887036869/job/27307936772) of https://github.com/conda-forge/pytorch-cpu-feedstock/pull/231/commits/8a59f3b7ba020c3e68e1becc1cc9c48af3fa2f29 and seeing
```
Encountered problems while solving:
  - package cuda-crt-12.2.140-h719f0c7_0 requires cuda-crt-dev_win-64 12.2.140.*, but none of the providers can be installed
```
Happy to hear if it's not necessary